server/container/guice/guice-common/src/main/java/org/apache/james/GuiceJamesServer.java

```diff
@@ -63,11 +66,18 @@ public GuiceJamesServer overrideWith(Module... overrides) {
     }
 
     public void start() throws Exception {
-        Injector injector = Guice.createInjector(module);
-        preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {}));
-        injector.getInstance(ConfigurationsPerformer.class).initModules();
-        guiceProbeProvider = injector.getInstance(GuiceProbeProvider.class);
-        isStarted = true;
+        try {
+            Injector injector = Guice.createInjector(module);
+            preDestroy = injector.getInstance(Key.get(new TypeLiteral<Stager<PreDestroy>>() {}));
+            injector.getInstance(ConfigurationsPerformer.class).initModules();
+            guiceProbeProvider = injector.getInstance(GuiceProbeProvider.class);
+            isStarted = true;
+        } catch (Exception e) {
+            LOGGER.error("Error while starting James. Stopping server.", e);
+            stop();
```

Comment:
I don't think the responsibility of  `start()` is to call `stop()`, I would rather do that at an upper layer or make the GuiceJamesServer object Closeable

Reply:
I don't see how Closeable does the trick (not working)

Reply:
I think you can agree it's not the responsibility of start to let a zombie server.

Reply:
If you mean doing it in the **main** classes, then it would lead to logic duplication.

Reply:
Ok, let me explain : 
1. you create your object
2. you run a method on it
if this method fails, you would like to have it revert the side effect of what it tried to do.

My opinion is : the start method fails, as a caller, it's your responsibility to know what you are doing with it. Maybe you want to retry ? Maybe you want to launch an analysis on what happened ? etc.

Basically, it's always the caller that decide when to cleanup an object. when a read method on a stream throws, it doesn't close it, right ?

Say GuiceJamesServer is closeable, the caller would write : 
```
try (server = new GuiceJamesServer) {
//do something with it
}
```